--- dbf.cpp.orig	2002-06-29 12:39:17.000000000 +0200
+++ dbf.cpp	2002-08-30 16:02:36.000000000 +0200
@@ -520,8 +520,12 @@
 #ifdef XB_MEMO_FIELDS
           s[i].Type != 'M' &&
 #endif /* XB_MEMO_FIELDS */
-          s[i].Type != 'L' )
+          s[i].Type != 'L' ) 
+      {
+        fclose( fp );
+        InitVars();
         xb_error(XB_UNKNOWN_FIELD_TYPE);
+      }
 
 #ifdef XB_MEMO_FIELDS
       if( !MemoSw && ( s[i].Type=='M' || s[i].Type=='B' || s[i].Type=='O'))
@@ -530,7 +534,11 @@
 
 // check for numeric fields which are too long
       if((s[i].Type == 'N' || s[i].Type == 'F') && s[i].FieldLen > 19 )
+      {
+        fclose( fp );
+        InitVars();
         xb_error(XB_INVALID_FIELD_LEN);
+      }
 
       i++;
    }
@@ -538,12 +546,14 @@
 
    if(( RecBuf = (char *) malloc( RecordLen )) == NULL )
    {
+      fclose( fp );
       InitVars();
       xb_memory_error;
    }
 
    if(( RecBuf2 = (char *) malloc( RecordLen )) == NULL ){
       free( RecBuf );
+      fclose( fp );
       InitVars();
       xb_memory_error;
    }
